src/Microsoft.Health.Dicom.Core/Features/Update/UpdateRequestValidator.cs

```diff
@@ -0,0 +1,85 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using EnsureThat;
+using FellowOakDicom;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Features.Validation;
+using Microsoft.Health.Dicom.Core.Messages.Update;
+using Microsoft.Health.Dicom.Core.Models.Update;
+
+namespace Microsoft.Health.Dicom.Core.Features.Update;
+
+/// <summary>
+/// Provides functionality to validate an <see cref="UpdateInstanceRequest"/>.
+/// </summary>
+public static class UpdateRequestValidator
+{
+    /// <summary>
+    /// Validates an <see cref="UpdateSpecification"/>.
+    /// </summary>
+    /// <param name="updateSpecification">The request to validate.</param>
+    /// <exception cref="BadRequestException">Thrown when request body is missing.</exception>
+    /// <exception cref="UidValidation">Thrown when the specified StudyInstanceUID is not a valid identifier.</exception>
+    public static void ValidateRequest(UpdateSpecification updateSpecification)
+    {
+        EnsureArg.IsNotNull(updateSpecification, nameof(updateSpecification));
+        if (updateSpecification.StudyInstanceUids == null || updateSpecification.StudyInstanceUids.Count == 0)
+        {
+            throw new BadRequestException(string.Format(CultureInfo.CurrentCulture, DicomCoreResource.MissingRequiredField, nameof(updateSpecification.StudyInstanceUids)));
+        }
+        foreach (var StudyInstanceUid in updateSpecification.StudyInstanceUids)
+        {
+            UidValidation.Validate(StudyInstanceUid, nameof(StudyInstanceUid), allowEmpty: true);
+        }
+    }
+
+    /// <summary>
+    /// Validates a <see cref="DicomDataset"/>.
+    /// </summary>
+    /// <param name="dataset">The Dicom dataset to validate.</param>
+    /// <exception cref="BadRequestException">Thrown when dicom tag or value validation fails</exception>
+    public static void ValidateDicomDataset(DicomDataset dataset)
+    {
+        EnsureArg.IsNotNull(dataset, nameof(dataset));
+        List<string> invalidTags = new List<string>();
+        List<string> invalidTagValues = new List<string>();
+        string errorString = string.Empty;
+
+        foreach (DicomItem item in dataset)
+        {
+            if (!UpdateTags.UpdateFilterTags.Contains(item.Tag))
+            {
+                invalidTags.Add(item.Tag.ToString());
+            }
+            try
+            {
+                item.Validate();
```

Comment:
If it is a invalid tag, do we need to validate the value?

Reply:
Yea, good catch. We dont have to right? I have moved the value validation to `else`